fundingmanager.go

```diff
@@ -2808,6 +2877,28 @@ func (f *fundingManager) handleInitFundingMsg(msg *initFundingMsg) {
 		return
 	}
 
+	// Check whether the peer supports upfront shutdown, and get an address which
+	// should be used (either a user specified address or a new address from the
+	// wallet if our node is configured to set shutdown address by default).
+	shutdown, err := getUpfrontShutdownScript(
+		msg.peer,
+		msg.openChanReq.shutdownScript,
+		func() (lnwire.DeliveryAddress, error) {
```

Comment:
perhaps extract this closure as a member of the funding manager? can this pass it to both as, e.g., `f.genDeliveryAddr`

Reply:
Peer already has a `genDeliveryScript`, so this (as is, and if added to funding manager) results in some duplication. An alternative would be to expose `genDeliveryScript` on the peer interface, or move all of the logic in `getUpfrontShutdownScript` into `f.genUpfrontShutdown`?

Reply:
I think it is just to avoid some clutter and indentation here, instead having the method be

```go
func(f *fundingmanager) genDeliveryAddr() (lnwire.DeliveryAddress, error) {
			addr, err := f.cfg.Wallet.NewAddress(lnwallet.WitnessPubKey, false)
			if err != nil {
				return nil, err
			}
			return txscript.PayToAddrScript(addr)
}
```